### PR TITLE
Random tangent vec in root class of Manifold and corresponding parametrized test in all manifolds (except product)

### PR DESCRIPTION
**Pull request by SaitejaUtpala**
Based on discussion with @nguigs  @ninamiolane. also tests are skipped for product manifold (shapes of it are not correct for it to work).

### CODE REVIEW

File: tests/tests_geomstats/test_euclidean.py [hunk 1]
```diff
@@ -61,6 +61,14 @@ def to_tangent_is_tangent_test_data(self):
                 self.n_vecs_list,
             )
 
+        def random_tangent_vec_is_tangent_test_data(self):
+            return self._random_tangent_vec_is_tangent_test_data(
+                Euclidean,
+                self.space_args_list,
+                self.n_vecs_list,
+                is_tangent_atol=gs.atol * 100,
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_euclidean.py:69]: How come we need an increased precision for the Euclidean space?

File: tests/tests_geomstats/test_general_linear.py [hunk 2]
```diff
@@ -298,6 +306,14 @@ def to_tangent_is_tangent_test_data(self):
                 self.n_vecs_list,
             )
 
+        def random_tangent_vec_is_tangent_test_data(self):
+            return self._random_tangent_vec_is_tangent_test_data(
+                SquareMatrices,
+                self.space_args_list,
+                self.n_vecs_list,
+                is_tangent_atol=gs.atol * 100,
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_general_linear.py:314]: ditto: why * 100?

File: tests/tests_geomstats/test_lower_triangular_matrices.py [hunk 3]
```diff
@@ -136,6 +136,14 @@ def to_tangent_is_tangent_test_data(self):
                 self.n_vecs_list,
             )
 
+        def random_tangent_vec_is_tangent_test_data(self):
+            return self._random_tangent_vec_is_tangent_test_data(
+                LowerTriangularMatrices,
+                self.space_args_list,
+                self.n_vecs_list,
+                is_tangent_atol=gs.atol * 100,
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_lower_triangular_matrices.py:144]: Ditto: Why * 100?

File: tests/tests_geomstats/test_matrices.py [hunk 4]
```diff
@@ -519,6 +519,14 @@ def basis_test_data(self):
             ]
             return self.generate_tests(smoke_data)
 
+        def random_tangent_vec_is_tangent_test_data(self):
+            return self._random_tangent_vec_is_tangent_test_data(
+                Matrices,
+                self.space_args_list,
+                self.n_vecs_list,
+                is_tangent_atol=gs.atol * 100,
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_matrices.py:527]: Ditto: why * 100?

File: tests/tests_geomstats/test_minkowski.py [hunk 5]
```diff
@@ -55,6 +55,14 @@ def to_tangent_is_tangent_test_data(self):
                 self.n_vecs_list,
             )
 
+        def random_tangent_vec_is_tangent_test_data(self):
+            return self._random_tangent_vec_is_tangent_test_data(
+                Minkowski,
+                self.space_args_list,
+                self.n_vecs_list,
+                is_tangent_atol=gs.atol * 100,
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_minkowski.py:63]: ditto

File: tests/tests_geomstats/test_symmetric_matrices.py [hunk 6]
```diff
@@ -144,6 +144,14 @@ def to_tangent_is_tangent_test_data(self):
                 self.n_vecs_list,
             )
 
+        def random_tangent_vec_is_tangent_test_data(self):
+            return self._random_tangent_vec_is_tangent_test_data(
+                SymmetricMatrices,
+                self.space_args_list,
+                self.n_vecs_list,
+                is_tangent_atol=gs.atol * 1000,
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_symmetric_matrices.py:152]: ?

**Review by ninamiolane — CHANGES_REQUESTED** [review]
LGTM, except some tolerance set to `gs.atol * 100` that I do not understand: e.g. in Euclidean space, why is this tolerance increased?

**Comment by SaitejaUtpala** [comment]
@ninamiolane . Sorry I was basically copy pasting data generation part. I will raise another PR now? If you can merge this that would be really great (It's just the it will take lot of time tests to run again) Thanks!

**Comment by SaitejaUtpala** [comment]
> @ninamiolane . Sorry I was basically copy pasting data generation part. I will raise another PR now? If you can merge this that would be really great (It's just the it will take lot of time tests to run again) Thanks!

Never mind. I will make change in this PR. Thanks!

File: tests/tests_geomstats/test_general_linear.py (hunk 2 above)
Comment (ninamiolane) [reply tests/tests_geomstats/test_general_linear.py:311]: Why *100?

**Review by ninamiolane — COMMENTED** [review]
The review comments do not seem to be addressed, e.g. why *100 in the square matrices?

**Comment by SaitejaUtpala** [comment]
> The review comments do not seem to be addressed, e.g. why *100 in the square matrices?

@ninamiolane  really sorry. I went in removed in all files but didn't check there are two manifolds in test_general_linear. Thanks for pointing out. Corrected it now

File: tests/tests_geomstats/test_symmetric_matrices.py (hunk 6 above)
Comment (ninamiolane) [reply tests/tests_geomstats/test_symmetric_matrices.py:152]: Why *1000 here?

**Review by ninamiolane — CHANGES_REQUESTED** [review]

**Pull request merged**